### ISSUE TITLE
Potential to support Hori PS4 controllers?

### ISSUE THREAD

**Issue by bud-cc** [Hydr8gon/VitaControl#14]
**Disclaimer:** _I'm not a dev so I have no idea what the level of effort is for adding support for additional controllers, so feel free to ELI5 and tell me no._

---
### Request:
Would it be possible to add support for the wireless Hori PS4-133(A), officially licensed PS4 controller?

---
### Controller:
Hori made two versions of this controller; wired and wireless. The wireless version was only available in the Japanese market.
 - [Hori **Wired** Mini Gamepad](https://stores.horiusa.com/wired-mini-gamepad-black/)  | [_link to manual_](https://drive.google.com/file/d/1xSL74QBDouysg2graDPdCe6CA-bdy8AA/view?pli=1)
 - [Hori **Wireless** Controller Light](https://hori.jp/products/p4/wlcl_c_bk/) | [_link to manual_](https://hori.jp/image/2019/01/ps4-133_manual_en.pdf)

The controllers lack a traditional touch pad, but instead have a "touch pad" button and a "touch pad mode" button. Details on how it operates are available in the manual links provided above.
```
Equipped with touch pad button / touch pad mode button

It is possible to simulate the touchpad/touchpad button of the wireless controller (DUALSHOCK®4).
```

The controllers also lack multiple functions present in Sony DS4 controllers like vibration, motion sensing, headphone jacks or speakers.
```
*This product does not have a vibration function, motion sensor function, stereo headphone/microphone terminal, or speaker.
```

I'm unsure if less features would mean less work, or just more complications.

---
### Testing:
I've attempted connection using all of usual DS4 controller plugins, which is how I found yours, and the results are always the same.
 - The controller is easily recognized and can be connected to via Settings>Devices>Bluetooth Devices
 - The controllers name entry is `HORI Wireless Controller Light`
 - The controller will connect but no commands sent from the controller are recognized at all.
 - Tested on my PC and it works fine.
 - FW is up to date per: ["Wireless Controller Light for PlayStation®4" firmware update notice](https://hori.jp/topix/update/p4-133/)

---
### Conclusion:
 - I feel like this would be the perfect companion controller for the Vita if it's possible to get it supported.
 - I don't care much about the touch pad functionality, if that makes it any simpler.
 - I only have one of these controllers available, but I'm willing to do any amount of leg work to help out.

**Comment by Hydr8gon** [comment]
I assume this controller works on PS4, so it's strange that it wouldn't work with the existing DS4 driver. The only thing I can suggest is to try rebooting after pairing, and letting the controller connect on its own; sometimes controllers don't work on the first pair for some reason.

Otherwise, supporting new controllers shouldn't be hard, as long as there's an existing open-source driver for reference. So far I haven't found anything for this particular controller, though.
